### ISSUE TITLE
`vmap` with indexing seems to be broken in `jax-metal`

### ISSUE THREAD

**Issue by saran-t** [google/jax#17175]
### Minimal example
```python
import jax
import jax.numpy as jp

x = jp.array([[1,2,3],[4,5,6],[7,8,9],[10,11,12]], dtype=jp.float32)

@jax.vmap
def f(i):
  return x[i]

f = jax.jit(f)
idx = jp.array([1,1,2,2,0])

f(idx)
```

### Expected result
```python
Array([[4., 5., 6.],
       [4., 5., 6.],
       [7., 8., 9.],
       [7., 8., 9.],
       [1., 2., 3.]], dtype=float32)
```

### What actually happens
The following exception is thrown:
```
XlaRuntimeError: UNKNOWN: <ipython-input-20-1307a3dff8de>:3:11: error: 'mps.reshape' op the result shape is not compatible with the input shape
<ipython-input-20-1307a3dff8de>:3:11: note: see current operation: %13 = "mps.reshape"(%11, %12) : (tensor<5xf32>, tensor<2xsi64>) -> tensor<5x3xf32>
```

The following debugging info is also dumped to my terminal when the error occurs. This doesn't seem to be part of the Python error message, but rather is something that `jax-metal` writes directly to stdout/stderr. Is there a way to disable this?
```"builtin.module"() ({
  "func.func"() ({
  ^bb0(%arg0: tensor<5xsi32>):
    %0 = "mps.constant"() {value = dense<0> : tensor<5x1xsi32>} : () -> tensor<5x1xsi32>
    %1 = "mps.constant"() {value = dense<4> : tensor<5xsi32>} : () -> tensor<5xsi32>
    %2 = "mps.constant"() {value = dense<0> : tensor<5xsi32>} : () -> tensor<5xsi32>
    %3 = "mps.constant"() {value = dense<[[1.000000e+00, 2.000000e+00, 3.000000e+00], [4.000000e+00, 5.000000e+00, 6.000000e+00], [7.000000e+00, 8.000000e+00, 9.000000e+00], [1.000000e+01, 1.100000e+01, 1.200000e+01]]> : tensor<4x3xf32>} : () -> tensor<4x3xf32>
    %4 = "mps.less"(%arg0, %2) : (tensor<5xsi32>, tensor<5xsi32>) -> tensor<5xi1>
    %5 = "mps.add"(%arg0, %1) : (tensor<5xsi32>, tensor<5xsi32>) -> tensor<5xsi32>
    %6 = "mps.select"(%4, %5, %arg0) : (tensor<5xi1>, tensor<5xsi32>, tensor<5xsi32>) -> tensor<5xsi32>
    %7 = "mps.constant"() {value = dense<[5, 1]> : tensor<2xsi64>} : () -> tensor<2xsi64>
    %8 = "mps.reshape"(%6, %7) : (tensor<5xsi32>, tensor<2xsi64>) -> tensor<5x1xsi32>
    %9 = "mps.constant"() {value = dense<1> : tensor<si32>} : () -> tensor<si32>
    %10 = "mps.concat"(%8, %0, %9) : (tensor<5x1xsi32>, tensor<5x1xsi32>, tensor<si32>) -> tensor<5x2xsi32>
    %11 = "mps.gather_nd"(%3, %10) {batch_dims = 0 : ui32} : (tensor<4x3xf32>, tensor<5x2xsi32>) -> tensor<5xf32>
    %12 = "mps.constant"() {value = dense<[5, 3]> : tensor<2xsi64>} : () -> tensor<2xsi64>
    %13 = "mps.reshape"(%11, %12) : (tensor<5xf32>, tensor<2xsi64>) -> tensor<5x3xf32>
    "func.return"(%13) : (tensor<5x3xf32>) -> ()
  }) {arg_attrs = [{jax.arg_info = "i", mhlo.sharding = "{replicated}"}], function_type = (tensor<5xsi32>) -> tensor<5x3xf32>, res_attrs = [{jax.result_info = ""}], sym_name = "main", sym_visibility = "public"} : () -> ()
}) {mhlo.num_partitions = 1 : i32, mhlo.num_replicas = 1 : i32, sym_name = "jit_f"} : () -> ()
```

### What jax/jaxlib version are you using?

jax v0.4.11, jaxlib v0.4.10 (built from source)

### Which accelerator(s) are you using?

Apple Silicon M1

### Additional system info

macOS 13.5 (22G74)

**Comment by skye** [comment]
@kulinseth @shuhand0 can you take a look?

cc @hawkinsp as well

**Comment by shuhand0** [comment]
We will look into the conversion pattern and add an option to enable/disable the dump.

**Comment by saran-t** [comment]
@shuhand0 thanks!

Unrelated, but since you're here: is there any plan to open source the plugin?

**Comment by shuhand0** [comment]
Thanks for asking. Binary release is our current solution, and we will update here if plan changes.

**Comment by benjaminvatterj** [comment]
This seems to be resolved as of the newest release (0.0.5)

**Comment by shuhand0** [comment]
The issue is fixed with jax-metal 0.0.5.
